### PR TITLE
refactor: remove built-in daemon manager

### LINKED ISSUE #2185: Proposal: remove built-in service daemon manager
# Proposal

## Please describe your feature

Remove built-in service daemon manager in backend that implement by go module `takama/daemon`.

Although this function exists, it brings some convenience. But it brings platform-related uncertainty to the operating system, makes it difficult to debug when problems occur during startup, and we need to add a lot of additional peripheral code to support it, such as writing out the PID file (when a program exception ends, the PID file will not be removed automatically correctly, which has caused many usage problems), the more irrelevant function code will cause more problems.

There are two main ways to use the new service management.
1. Container
At this point, the life cycle of the manager API is fully managed to the container runtime, without manual user processing.

2. RPM & DEB installation package
At this point, we will use the systemd service file to register the APISIX Dashboard service with systemd management, which users can manage using `service` or `systemctl'.

3. Or run it directly.
When the user uses the Manager API for debugging purposes, the user can run it directly through the shell, and all the output will be printed in stdout and stderr. Or use `nohup` to run it in the background.

## Describe the solution you'd like

- [ ] Remove code about service management.
- [ ] Add the dashboard service code in the `apisix-build-tools` tool repository.

### PR DESCRIPTION
**Pull request by bzp2010**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [ ] New feature provided
- [x] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Remove built-in service daemon manager and move to systemd service unit.

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [x] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Review by iamayushdas — APPROVED** [review]

**Review by TakumiInui5432 — APPROVED** [review]

**Review by iamayushdas — APPROVED** [review]

**Review by iamayushdas — APPROVED** [review]

**Comment by bzp2010** [comment]
### Update - 2021-11-01
- [x] remove built-in service manager
- [x] refactor cli tests and change to use systemd service manager
- [x] provide new deploy documentation

ping @starsz @nic-chen for code review
ping @juzhiyuan @liuxiran for docs review

File: docs/en/latest/install.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+           apache/apisix-dashboard
+```
+
+:::note
+Please replace `<CONFIG_FILE>` to your configure file path.
+:::
+
+## RPM {#rpm}
+
+**NOTE:** Only support CentOS 7 currently, for more information, please refer to [here](./deploy.md).
```
Comment (liuxiran) [inline docs/en/latest/install.md:45]:
```suggestion
**NOTE:** Only CentOS 7 is supported currently, for more information, please refer to [here](./deploy.md).
```

File: docs/en/latest/install.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+[Unit]
+Description=apisix-dashboard
+Conflicts=apisix-dashboard.service
+After=network-online.target
+
+[Service]
+WorkingDirectory=/usr/local/apisix-dashboard
+ExecStart=/usr/local/apisix-dashboard/manager-api -c /usr/local/apisix-dashboard/conf/conf.yaml
+```
+
```
Comment (liuxiran) [inline docs/en/latest/install.md:144]: May I ask what is the system command when I want to use this way

**Review by liuxiran — APPROVED** [review]
Basic Deploy description is LGTM

**Review by juzhiyuan — APPROVED** [review]

File: api/cmd/stop.go [hunk 3]
```diff
@@ -1,54 +0,0 @@
...
-
-	"github.com/spf13/cobra"
-
-	"github.com/apisix/manager-api/internal/conf"
-	"github.com/apisix/manager-api/internal/utils"
-)
-
-func newStopCommand() *cobra.Command {
-	return &cobra.Command{
-		Use:   "stop",
```
Comment (nic-chen) [inline api/cmd/stop.go:32]: wow, We should keep the `stop` command, it has nothing to do with the daemon.

Comment (bzp2010) [reply api/cmd/stop.go:32]: @nic-chen The original code looked for the specified process by the PID file, which proved unreliable by many bug reports, and this part has been removed. Therefore, the original `stop` operation is not well supported.

**Review by starsz — APPROVED** [review]

**Pull request merged**